ui/js/dfv/src/fields/number-field/index.js

```diff
@@ -1,33 +1,131 @@
-	/* Enable this only for slider
-	props.validation.addRules( [
-		{
-			rule: validationRules.max( props.value, props.fieldConfig.number_max ),
-			condition: true,
-		},
-		{
-			rule: validationRules.min( props.value, props.fieldConfig.number_min ),
-			condition: true,
-		},
-	] );*/
-
-	// noinspection JSUnresolvedVariable
+import { debounce } from 'lodash';
+import PropTypes from 'prop-types';
+
+import numberFormat from 'dfv/src/helpers/numberFormat';
+import { toBool } from 'dfv/src/helpers/booleans';
+import { FIELD_PROP_TYPE_SHAPE } from 'dfv/src/config/prop-types';
+
+import './number-field.scss';
+
+const formatValue = ( newValue, decimals, format ) => {
+	if ( 'undefined' === typeof newValue ) {
+		newValue = 0;
+	}
+
+	let thousands = '';
+	let dot = '.';
+
+	// @todo we need to localize a value from PHP to get the wp_locale values
+	// as the defaults, something similar to:
+	// https://github.com/WordPress/gutenberg/blob/9ced21b344695052db1b3fd10a865b205e5c418e/lib/compat.php#L264-L303
+	//
+	// thousands = wp_locale['number_format']['thousands_sep'];
+	// dot       = wp_locale['number_format']['decimal_point'];
+
+	switch ( format ) {
+		case '9.999,99':
+			thousands = '.';
+			dot = ',';
+			break;
+		case '9,999.99':
+			thousands = ',';
+			dot = '.';
+			break;
+		case "9'999.99":
+			thousands = '\'';
+			dot = '.';
+			break;
+		case '9 999,99':
+			thousands = ' ';
+			dot = ',';
+			break;
+		case '9999.99':
+			thousands = '';
+			dot = '.';
+			break;
+		case '9999,99':
+			thousands = '';
+			dot = ',';
+			break;
+		case 'i18n':
+			// fall through to default
+		default:
+			break;
+	}
+
+	return numberFormat( parseFloat( newValue ), decimals, dot, thousands );
+};
+
+const NumberField = ( {
+	fieldConfig,
+	value,
+	setValue,
+} ) => {
+	const {
+		htmlAttr: htmlAttributes = {},
+		readonly: readOnly,
+		number_decimals: decimals,
+		number_format: format,
+		number_format_soft: softFormat,
+		number_format_type: type,
+		number_max: max,
+		number_max_length: maxLength,
+		number_min: min,
+		number_placeholder: placeholder,
+		number_step: step,
+	} = fieldConfig;
+
+	const correctedDecimals = toBool( softFormat ) ? 0 : decimals;
+
+	const handleChange = ( event ) => debounce( () => {
+		setValue(
+			formatValue( event.target.value, correctedDecimals, format )
+		);
+	}, 1000 );
+
+	if ( 'number' === type ) {
+		return (
+			<input
+				type="text"
+				name={ htmlAttributes.name }
+				id={ htmlAttributes.id }
+				placeholder={ placeholder }
+				maxLength={ -1 !== parseInt( maxLength, 10 ) ? maxLength : undefined }
+				value={ value }
+				readOnly={ !! readOnly }
+				onChange={ handleChange }
+			/>
+		);
+	}
+
 	return (
-		<BaseInput
-			type={ '1' === props.fieldConfig.number_html5 ? 'number' : 'text' }
-			//min={props.fieldConfig.number_min} Enable this only for slider
-			//max={props.fieldConfig.number_max} Enable this only for slider
-			{ ...props }
-		/>
+		<div>
```

Comment:
This should only happen if the `slider` type is set, otherwise it should always fallback to the regular number field, let's move this to the if conditional above and let the regular number field be the final return